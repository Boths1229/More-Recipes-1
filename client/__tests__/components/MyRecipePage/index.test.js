import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  MyRecipePage
} from '../../../components/MyRecipePage/index';
import MockData from '../../__mocks__/actions/recipes';

let props;

const setup = () => {
  props = {
    user: MockData.userData.profile,
    actions: {
      logoutAction: () => {},
      getUserRecipes: () => {},
      deleteRecipeAction: () => Promise.resolve(),
      editRecipeAction: () => Promise.resolve(),
    },
    userRecipes: MockData.userRecipesSuccess.userRecipes,
    isLoading: false,
    pages: MockData.userRecipesSuccess.pages
  };

  return shallow(<MyRecipePage {...props} />);
};

localStorage.clear = () => {};
localStorage.getItem = () => {};

describe('MyRecipePage Component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render message if user have no recipe', () => {
    const wrapper = setup();
    wrapper.setProps({ userRecipes: [] });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text()).toBe('You do not have any recipe');
  });

  it('should delete recipe when the delete button is clicked', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const spy = jest.spyOn(action, 'handleDeletion');
    action.handleDeletion();
    expect(spy).toBeCalled();
  });

  it('should submit edit recipe form', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const spy = jest.spyOn(action, 'handleSubmit');
    action.handleSubmit({ preventDefault: () => {} });
    expect(spy).toBeCalled();
  });


  it('should set state of input field', () => {
    const event = {
      target: { name: 'recipeName', value: 'Rice' }
    };
    const wrapper = setup();
    const action = wrapper.instance();
    const spy = jest.spyOn(action, 'handleChange');
    action.handleChange(event);
    expect(spy).toBeCalled();
    expect(wrapper.state('recipe').recipeName).toBe('Rice');
  });


  it('should clear error message when form input is focused on', () => {
    const wrapper = setup();
    wrapper.setState({ errorMessage: 'Invalid' });
    const action = wrapper.instance();
    const spy = jest.spyOn(action, 'onFocus');
    action.onFocus();
    expect(spy).toBeCalled();
    expect(wrapper.state('errorMessage')).toBe('');
  });
});
