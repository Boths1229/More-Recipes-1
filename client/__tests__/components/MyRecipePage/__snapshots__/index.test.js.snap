// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipePage Component should render correctly 1`] = `
<div>
  <Connect(authHeader)
    user={
      Object {
        "createdAt": "2018-02-19T15:54:05.322Z",
        "email": "tester@test.com",
        "id": 1,
        "image": null,
        "username": "sannytee",
      }
    }
  />
  <div
    className="container"
  >
    <div
      className="row section-popular"
    >
      <div
        className="col-sm-12"
      >
        <h2
          className="pt-3 headline centered"
        >
          My Recipes
        </h2>
      </div>
    </div>
    <div
      className="row pt-4"
    />
    <EditRecipeModal
      details={
        Object {
          "recipeName": "",
        }
      }
      errorMessage=""
      handleChange={[Function]}
      handleSubmit={[Function]}
      isUploading={false}
      onFocus={[Function]}
      progress={0}
      startUpload={[Function]}
      upload={[Function]}
    />
    <DeleteRecipeModal
      handleDeletion={[Function]}
    />
  </div>
  <div
    className="sticky-paginate"
  >
    <Paginate
      handlePaginationChange={[Function]}
      page={2}
    />
  </div>
  <Footer />
</div>
`;

exports[`MyRecipePage Component should render message if user have no recipe 1`] = `
<div>
  <Connect(authHeader)
    user={
      Object {
        "createdAt": "2018-02-19T15:54:05.322Z",
        "email": "tester@test.com",
        "id": 1,
        "image": null,
        "username": "sannytee",
      }
    }
  />
  <div
    className="container"
  >
    <div
      className="row section-popular"
    >
      <div
        className="col-sm-12"
      >
        <h2
          className="pt-3 headline centered"
        >
          My Recipes
        </h2>
      </div>
    </div>
    <div
      className="row pt-4"
    >
      <h1
        style={
          Object {
            "margin": "150px auto 0",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        You do not have any recipe
      </h1>
    </div>
    <EditRecipeModal
      details={
        Object {
          "recipeName": "",
        }
      }
      errorMessage=""
      handleChange={[Function]}
      handleSubmit={[Function]}
      isUploading={false}
      onFocus={[Function]}
      progress={0}
      startUpload={[Function]}
      upload={[Function]}
    />
    <DeleteRecipeModal
      handleDeletion={[Function]}
    />
  </div>
  <div />
  <Footer />
</div>
`;
